### PR TITLE
Rename chunksOf and arraysOf

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/2249

File: core/src/Streamly/Internal/Data/Fold.hs [hunk 1]
```diff
@@ -283,7 +283,7 @@ module Streamly.Internal.Data.Fold
     -- ** Splitting
     , many
     , manyPost
-    , chunksOf
+    , foldChunksOf
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Fold.hs:286]: We can use the name `groupsOf` instead.

**Review by harendra-kumar — COMMENTED** [review]
Since chunksOf is a released API, let's deprecate it and add `groupsOf` instead. keep ararysOf as it is for now, it is an internal API.

File: core/src/Streamly/Data/Fold.hs [hunk 2]
```diff
@@ -325,7 +325,7 @@ module Streamly.Data.Fold
 
     -- ** Splitting
     , many
-    , chunksOf
+    , groupsOf
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Fold.hs:328]: chunksOf cannot be removed, it should be deprecated.

File: core/src/Streamly/Internal/Data/Array/Mut/Stream.hs [hunk 3]
```diff
@@ -55,7 +55,6 @@ import qualified Streamly.Internal.Data.Parser.ParserD as ParserD
 arraysOf :: (MonadIO m, Unbox a)
     => Int -> Stream m a -> Stream m (MutArray a)
 arraysOf = MArray.arraysOf
-
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/Array/Mut/Stream.hs:58]: Add this line back.

File: core/src/Streamly/Internal/Data/Array/Mut/Type.hs [hunk 4]
```diff
@@ -1295,9 +1295,6 @@ data GroupState s contents start end bound
 arraysOf :: forall m a. (MonadIO m, Unbox a)
     => Int -> D.Stream m a -> D.Stream m (MutArray a)
 -- XXX the idiomatic implementation leads to large regression in the D.reverse'
--- benchmark. It seems it has difficulty producing optimized code when
--- converting to StreamK. Investigate GHC optimizations.
--- arraysOf n = D.foldMany (writeN n)
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/Array/Mut/Type.hs:1300]: Why remove this?

File: core/src/Streamly/Internal/Data/Stream/Chunked.hs [hunk 5]
```diff
@@ -111,7 +111,7 @@ import qualified Streamly.Internal.Data.Stream.StreamK as K
 -- | @arraysOf n stream@ groups the elements in the input stream into arrays of
 -- @n@ elements each.
 --
--- > arraysOf n = Stream.chunksOf n (Array.writeN n)
+-- > arraysOf n = Stream.arraysOf n (Array.writeN n)
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/Stream/Chunked.hs:114]: Shouldn't this be `groupsOf`?

File: core/src/Streamly/Internal/FileSystem/Handle.hs [hunk 6]
```diff
@@ -494,17 +494,17 @@ writeChunksWithBufferOf = writeChunksWith
 -- do not want buffering to occur at GHC level as well. Same thing applies to
 -- writes as well.
 
--- XXX Maybe we should have a Fold.arraysOf like we have Stream.arraysOf
+-- XXX Maybe we should have a Fold.groupsOf like we have Stream.groupsOf
```
Comment (adithyaov) [inline core/src/Streamly/Internal/FileSystem/Handle.hs:497]: This should not be changed? `arraysOf` and `groupsOf` is different?

File: src/Streamly/Internal/Data/Stream/IsStream/Reduce.hs [hunk 7]
```diff
@@ -970,7 +970,7 @@ splitOnSuffixSeqAny _subseq _f _m = undefined
 -- | Group the input stream into groups of @n@ elements each and then fold each
 -- group using the provided fold function.
 --
--- >>> Stream.toList $ Stream.chunksOf 2 Fold.sum (Stream.enumerateFromTo 1 10)
+-- >>> Stream.toList $ Stream.arraysOf 2 Fold.sum (Stream.enumerateFromTo 1 10)
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Stream/IsStream/Reduce.hs:973]: Shouldn't this be `groupsOf`?

**Review by adithyaov — CHANGES_REQUESTED** [review]
Some comments.

**Comment by adithyaov** [comment]
https://github.com/composewell/streamly/issues/2292

File: benchmark/Streamly/Benchmark/FileSystem/Handle/Read.hs [hunk 8]
```diff
@@ -231,34 +231,34 @@ foldManyChunksOfSum n inh =
...
-inspect $ 'chunksOf `hasNoType` ''IUF.ConcatState -- FH.read/UF.many
+inspect $ 'groupsOf `hasNoType` ''IUF.ConcatState -- FH.read/UF.many
 #endif
 
-{-# INLINE arraysOf #-}
-arraysOf :: Int -> Handle -> IO Int
-arraysOf n inh =
+{-# INLINE chunksOf #-}
+chunksOf :: Int -> Handle -> IO Int
+chunksOf n inh =
```
Comment (adithyaov) [inline benchmark/Streamly/Benchmark/FileSystem/Handle/Read.hs:252]: See https://github.com/composewell/streamly/issues/2292

File: benchmark/Streamly/Benchmark/FileSystem/Handle/Read.hs [hunk 9]
```diff
@@ -281,20 +281,20 @@ o_1_space_reduce_read_grouped env =
...
-
-        -- arraysOf may use a different impl than chunksOf
+        , mkBenchSmall "S.groupsOf 1" env $ \inh _ ->
+            groupsOf 1 inh
+        , mkBench "S.groupsOf 10" env $ \inh _ ->
+            groupsOf 10 inh
+        , mkBench "S.groupsOf 1000" env $ \inh _ ->
+            groupsOf 1000 inh
+
+        -- chunksOf may use a different impl than groupsOf
```
Comment (adithyaov) [inline benchmark/Streamly/Benchmark/FileSystem/Handle/Read.hs:291]: This comment is incorrect.

**Comment by adithyaov** [comment]
See https://github.com/composewell/streamly/pull/2293

**Comment by adithyaov** [comment]
See https://github.com/composewell/streamly/pull/2293

**Pull request closed without merging**